File: 3024/Mountain Ranges.cpp
using namespace std;
#include <iostream>
#include <vector>
#include<bits/stdc++.h>
#include <algorithm>
#include <string>


int main(){

    int n, x, k;
    int maxi = 0, count = 0;
    vector < pair<int, bool> > flags;
    cin >> n >> x;
    
    for (int i = 0; i < n; i ++){
        cin >> k;
        flags.push_back(make_pair(k,false));
    }
    
    for (int i = 0; i < n-1; i++){
        int value = flags[i+1].first - flags[i].first;
        //out << value << endl;
        if (value <= x){
            if (flags[i+1].second == false){
                count++;
                flags[i+1].second = true;
            }
            if (flags[i].second == false){
                count++;
                flags[i].second = true;
            }
        }else{
            if(count>=maxi){
                maxi = count;
            }
            count = 0;
        }
    }
    
    if (count >= maxi){
        maxi = count;
    }
    
    if (n == 1)
        cout << 1 << endl;
    else
        cout << maxi << endl;
    
    
}

File: 3024/Mountain Ranges.py
n, f = map (int, input().split())

numbers = input().split()
flags = []
for x in range(len(numbers)):
    flags.append([int(numbers[x]), False])
maxi = 0
count = 0
for x in range(0, len(numbers)-1):
    value = flags[x+1][0] - flags[x][0]
    #print(value)
    if value <= f:
        if flags[x+1][1] == False:
            count+=1
            flags[x+1][1] = True
        if flags[x][1] == False:
            count+=1
            flags[x][1] = True

        
    
    else:
        if count >= maxi:
            maxi = count
        count = 0
if count >= maxi:
    maxi = count

if len(numbers)==1:
    print(1)
else:

    print(maxi)
